gluten-core/src/main/scala/io/glutenproject/expression/ScalarSubqueryTransformer.scala

```diff
@@ -41,4 +47,16 @@ class ScalarSubqueryTransformer(plan: BaseSubqueryExec, exprId: ExprId,
     }
     ExpressionBuilder.makeLiteral(result, query.dataType, result == null)
   }
+
+  /**
+   * DataTypes which supported or not depend on actual value
+   *
+   * @param dataType
+   * @return
+   */
+  def valueSensitiveDataType(dataType: DataType): Boolean = {
+    dataType.isInstanceOf[MapType] ||
+      dataType.isInstanceOf[ArrayType] ||
+      dataType.isInstanceOf[StructType]
+  }
```

Comment:
Hi @loudongfeng , I just bumped into this code. Do you still remember the reason of using this condition?